pkg/utils/k8sutils.go

```diff
@@ -171,3 +175,35 @@ func IsForbiddenError(err error) bool {
 	}
 	return false
 }
+
+func SetClusterInformation(
+	ctx context.Context,
+	kubeClient client.Client,
+	status interfaces.StorageClientInfo,
+) error {
+
+	clusterVersion := &configv1.ClusterVersion{}
+	clusterVersion.Name = "version"
+	if err := kubeClient.Get(ctx, client.ObjectKeyFromObject(clusterVersion), clusterVersion); err != nil {
+		return fmt.Errorf("failed to get cluster version: %v", err)
+	}
+	historyRecord := Find(clusterVersion.Status.History, func(record *configv1.UpdateHistory) bool {
+		return record.State == configv1.CompletedUpdate
+	})
+	if historyRecord == nil {
+		return fmt.Errorf("unable to find the updated cluster version")
+	}
+	status = status.
+		SetClientPlatformVersion(historyRecord.Version).
+		SetClusterID(string(clusterVersion.Spec.ClusterID))
```

Comment:
You are pulling the cluster version on line 189, can you please move the setClusterID function there?

Reply:
done.